sdk/runanywhere-kotlin/src/commonMain/kotlin/com/runanywhere/sdk/public/extensions/RunAnywhereModelLifecycle.kt

```
@@ -0,0 +1,202 @@
+    try {
+        // Perform unload based on modality
+        when (modality) {
+            Modality.LLM, Modality.VLM -> {
+                com.runanywhere.sdk.foundation.ServiceContainer.shared
+                    .modelLoadingService.unloadModel(state.modelId)
+            }
+            else -> {
+                // STT, TTS, and other components handle their own cleanup
+            }
+        }
+    } catch (e: Exception) {
```

detekt (1.23.8)

[warning] The caught exception is swallowed. The original exception could be lost.

(detekt.exceptions.SwallowedException)